.github/workflows/superset-e2e.yml

```diff
@@ -73,6 +73,7 @@ jobs:
         with:
           persist-credentials: false
           submodules: recursive
+          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha }}
```

Comment:
nastiest github bug I've ever seen, `actions/checkout` somehow sometimes decides to checkout random commit -> https://github.com/actions/checkout/issues/299#issuecomment-2881779075